syntax = "proto3";

package cs739;


service RBS {
  rpc CheckPrimary(CheckPrimaryRequest) returns(Response);
  rpc Read(ReadRequest) returns(Response);
  rpc Write(WriteRequest) returns(Response);
}

message CheckPrimaryRequest {

}

message ReadRequest {
  int64 address = 1;
}

message WriteRequest {
  int64 address = 1;
  string data = 2;
}

message Response {
  int32 return_code = 1;
  int32 error_number = 2;
  int32 primary = 3;
  string data = 3;
}